src/app/link-list/link-list.component.html

```diff
@@ -0,0 +1,12 @@
+<div class="search-bar">
+    <input #searchBox (input)="search(searchBox.value)" 
+            placeholder="Insert your regular expression here"
+            autofocus="autofocus"/>
+</div>
+<div class="links">    
+    <ul>
+        <li *ngFor="let link of links$ | async">
```

Comment:
This is a great example where a custom pipe function can be used.
```
<li *ngFor="let link of links$ | filterByRegex">
```
Then you need to implement something like 
```
@Pipe({name: 'filterByRegex'})
export class FilterPipe implements PipeTransform {
...
}
```